File: src/main/java/us/illyohs/civilmagiks/common/core/capability/handler/ManaHandler.java
package us.illyohs.civilmagiks.common.core.capability.handler;

import net.minecraft.nbt.NBTTagCompound;
import net.minecraftforge.common.util.INBTSerializable;
import us.illyohs.civilmagiks.api.capability.IManaHandler;
import us.illyohs.civilmagiks.api.mana.ManaType;

public class ManaHandler implements IManaHandler, INBTSerializable<NBTTagCompound> {

    @Override
    public NBTTagCompound serializeNBT() {
        return null;
    }

    @Override
    public void deserializeNBT(NBTTagCompound nbt) {

    }

    @Override
    public int getMaxStorage() {
        return 0;
    }

    @Override
    public int getMana() {
        return 0;
    }

    @Override
    public ManaType getManaType() {
        return null;
    }

    @Override
    public boolean canAcceptMana() {
        return false;
    }
}

File: src/main/scala/us/illyohs/civilmagiks/common/core/capability/handler/ManaHandler.scala
package us.illyohs.civilmagiks.common.core.capability.handler

import us.illyohs.civilmagiks.api.capability.IManaHandler
import us.illyohs.civilmagiks.api.init.ManaTypes
import us.illyohs.civilmagiks.api.mana.Mana

class ManaHandler extends IManaHandler {

  private var currentMana: Int = 0
  private var maxMana: Int = 0
  private var manaType: Mana = null

  def setMaxMana(maxMana: Int) {
    this.maxMana = maxMana
  }

  def addMana(add: Int) {
    if (getCurrentMana + add > maxMana) {
      this.currentMana = getCurrentMana + 0
    }
    else if (getCurrentMana + add <= maxMana) {
      this.currentMana = getCurrentMana + add
    }
  }

  def getMaxStorage: Int = {
    maxMana
  }

  def getCurrentMana: Int = {
    currentMana
  }

  def setType(mana: Mana) {
    this.manaType = mana
  }

  def getManaType: Mana = {
    ManaTypes.GREY
  }
}